addons/survey/static/src/js/survey_form.js

```diff
@@ -260,7 +261,18 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         } else if (target.value === 'next_skipped') {
             options.nextSkipped = true;
         } else if (target.value === 'finish') {
-            options.isFinish = true;
+            // Adding pop-up before the survey is submitted
+            this.call("dialog", "add", ConfirmationDialog, {
+                title: _t("Submit confirmation"),
+                body: _t("Are you sure you want to submit the survey?"),
+                confirmLabel: _t("Submit"),
+                confirm: () => {
+                    options.isFinish = true;
+                    this._submitForm(options);
+                },
+                cancel: () => {},
+            });
+            return;
         }
         this._submitForm(options);
```

Comment:
I would prefer if we copied this line inside the 2 other if cases, small detail but it's a bit easier to read than a random early return at the bottom of a function. 

You could even rework it a bit and do:

`this._submitForm({previousPageId: parseInt(target.dataset['previousPageId'])});`
`this._submitForm({nextSkipped: true)});`
`this._submitForm({isFinish: true)});`

And get rid of the options variable.